apps/server/src/modules/tool/service/tool-validation.service.spec.ts

```diff
@@ -293,6 +327,30 @@ describe('ToolValidation', () => {
 			});
 		});
 
+		describe('validateDefaultValue', () => {
+			describe('when default value matches regex', () => {
+				it('should not throw', async () => {
+					const externalToolDO: ExternalToolDO = externalToolDOFactory
+						.withCustomParameters(1, { default: 'test', regex: '[t]', regexComment: 'mockComment' })
+						.buildWithId();
+
+					const result: Promise<void> = service.validateUpdate(externalToolDO.id as string, externalToolDO);
+
+					await expect(result).resolves.not.toThrow();
+				});
+
+				it('should throw', async () => {
```

Comment:
same here: "when default value matches regex, it should not throw, and it should throw".